### PR TITLE
Improve error messaging

### PR DESCRIPTION
**Pull request by ravinagill15**
## In this PR:
- Added call data to `CreditFailed` error to help with debugging 
- See conversation [here](https://github.com/LayerZero-Labs/devtools/pull/838#discussion_r1738895442)

### CODE REVIEW

**Comment by ravinagill15** [comment]
Will add changeset if this is merged after `feat/native-oft-adapter`

File: packages/oft-evm/contracts/NativeOFTAdapter.sol [hunk 1]
```diff
@@ -112,7 +112,7 @@ abstract contract NativeOFTAdapter is OFTCore {
         // @dev Transfer tokens to the recipient.
         (bool success, ) = payable(_to).call{value: _amountLD}("");
         if (!success) {
-            revert CreditFailed();
+            revert CreditFailed(_to, _amountLD);
```
Comment (janjakubnanista) [inline packages/oft-evm/contracts/NativeOFTAdapter.sol:115]: Oh in my comment I meant something like this:

```solidity
error CreditFailed(address to, bytes revertData);

// If the call fails, `data` will hold the revert reason
(bool success, bytes memory data) = payable(_to).call{value: _amountLD}("");
if (!success) revert CreditFailed(_to, data);
```

Thoughts and prayers please @TRileySchwarz

Comment (ravinagill15) [reply packages/oft-evm/contracts/NativeOFTAdapter.sol:115]: Oh i see! That would definitely be helpful -- I will update this in the next commit 😄

Comment (ravinagill15) [reply packages/oft-evm/contracts/NativeOFTAdapter.sol]: I kept the `to` and `amountLD` parameters bc I think they could be helpful as well and also added the `revertData` as a third parameter now

**Review by TRileySchwarz — APPROVED** [review]

**Pull request merged**